File: backend/app/schemas/operation_log.py
"""业务操作日志 API 模型。"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class OperationLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: str
    entity_type: str
    entity_id: str
    entity_number: str | None = None
    action: str
    summary: str
    actor_user_id: str | None = None
    actor_email: str | None = None
    changes: dict = Field(default_factory=dict)
    metadata: dict = Field(default_factory=dict, validation_alias="metadata_json")
    created_at: datetime


class OperationLogListResponse(BaseModel):
    items: list[OperationLogRead]
    total: int

File: src/lib/api/operationLog.ts
import { apiGet } from '@/lib/api/client.ts'

export type OperationLogChangeValue = {
  before?: unknown
  after?: unknown
}

export type OperationLogRow = {
  id: string
  entity_type: 'rd' | 'channel' | string
  entity_id: string
  entity_number?: string | null
  action: string
  summary: string
  actor_user_id?: string | null
  actor_email?: string | null
  changes: Record<string, OperationLogChangeValue>
  metadata?: Record<string, unknown>
  created_at: string
}

export type OperationLogListResponse = {
  items: OperationLogRow[]
  total: number
}

export function listOperationLogs(params: {
  entity_type?: string
  entity_id?: string
  action?: string
  include_related?: boolean
  limit?: number
  offset?: number
} = {}): Promise<OperationLogListResponse> {
  const query = new URLSearchParams()
  if (params.entity_type) query.set('entity_type', params.entity_type)
  if (params.entity_id) query.set('entity_id', params.entity_id)
  if (params.action) query.set('action', params.action)
  if (params.include_related) query.set('include_related', 'true')
  if (params.limit != null) query.set('limit', String(params.limit))
  if (params.offset != null) query.set('offset', String(params.offset))
  const qs = query.toString()
  return apiGet<OperationLogListResponse>(`/api/operation-logs${qs ? `?${qs}` : ''}`)
}
